src/log/Log.cc

```diff
@@ -184,6 +186,25 @@ Entry *Log::create_entry(int level, int subsys)
   }
 }
 
+Entry *Log::create_entry(int level, int subsys, size_t* expected_size)
+{
+  if (true) {
+    size_t size = __atomic_load_n(expected_size, __ATOMIC_RELAXED);
+    void* ptr=malloc(sizeof(Entry) + size);
```

Comment:
and delete is used elsewhere to free.  just switch this to new char[...] and switch the deletes to delete[]?